src/main/java/com/spring/demo/controllers/DemoController.java

```diff
@@ -14,30 +18,42 @@
 
 @RestController
 @RequestMapping("/api")
+@CrossOrigin
 public class DemoController {
 
+  private final ItemGateway itemGateway;
+
+  @Autowired
+  public DemoController(ItemServiceDiskImpl itemServiceDisk, ItemGateway itemGateway) {
+    this.itemGateway = itemGateway;
+  }
+
   @GetMapping(path = "/items")
   public List<Item> getItems(@RequestParam(required = false) String id) {
-    return Arrays.asList(new Item());
+    if (id == null || id.isBlank() || id.isEmpty()) {
```

Comment:
using util method to check empty input
`org.apache.commons.lang3.StringUtils.isEmpty`